### PR TITLE
feat(curriculum): create css variable quiz

### LINKED ISSUE #56193: [HACKTOBERFEST 2024] - CSS Variables Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the CSS Variables quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-css-variables`.

**All quiz questions need to relate to one of these topics:**

- custom properties
  - defining variables (--foreground)
  - using variables (var(--foreground))
- `@property` rule

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by Ayushjhawar8**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56193

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:47]:
````suggestion
In the following code, which value will color have if --main-color is not defined?

```css
p {
  color: var(--main-color, pink);
}
```
````

code fences need to be surrounded by empty lines and they need the language name to know which highlighting to use. please check your other code fences and fix this

Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: committed the change you suggested, is it fine now?

Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: You successfully fixed this issue, yes

**Comment by ilenia-magoni** [comment]
please do not merge main into your branch unless asked to/unless you have merge conflicts

**Comment by Ayushjhawar8** [comment]
oh okay i didnt knew that sorry, currently is it fine or created issue?

**Comment by ilenia-magoni** [comment]
if maintainers had allowed tests for this PR, each time you do a commit, including a merge commit, the tests run, to avoid clogging the tests queue avoiding merging unless necessary is a good practice

**Comment by Ayushjhawar8** [comment]
okay i just found that out...will not do same mistake again

**Review by naomi-lgbt — APPROVED** [review]
Nice work! This looks ready to go!

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:20]: I'm not actually certain a user can set that in their browser. Unless I'm missing something, in which case provide a link so I can read up on it. 

https://developer.mozilla.org/en-US/docs/Web/CSS/animation-duration

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:86]: This doesn't seem quite right to me. https://developer.mozilla.org/en-US/docs/Web/CSS/--* 

Mind elaborating?

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:20]: You're right that not all browsers have a direct setting for adjusting animation speed. However, many modern operating systems and browsers support a "reduce motion" option in their accessibility settings. This allows users to minimize or disable animations for a more comfortable experience.

You can also check user preferences with the `prefers-reduced-motion` media query in CSS. If a user has this setting enabled, you can adjust your animations accordingly.

https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion
https://web.dev/articles/prefers-reduced-motion

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:86]: yea i got your point..correcting the answer now it should be `--background-color: #333;`

**Comment by Ayushjhawar8** [comment]
um commit message i think is wrong linking to an issue `#333`

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:108]: i appreciate your review...after i re read the question the most suitable answer seems to be `To define reusable styles across a project` not `to change the DOM Structure`  

fixing it now as well

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:422]: There's previous very similar quesion:
> When defining a variable `--foreground` inside a media query, what happens when the media query no longer matches the current viewport?
> - The custom property is preserved and continues to apply
> - The custom property reverts to its initial value
> - The custom property is recalculated based on the viewport
> - The custom property is no longer available as it is scoped to the media query

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:448]: This is the same as correct answer, distractors must be different.

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:334]: There's already identical question used.

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: resolved

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: resolved in new commit

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: resolved in new commit

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: resolved in new commit

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: resolved in new commit

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: resolved in new commit

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: resolved in new commit (changed question)

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: resolved in new commit (changed question)

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: resolved in new commit

**Review by majestic-owl448 — COMMENTED** [review]
So it seems I had a review pending since a while ago

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:302]: okay i changed it to an more incorrect option so it does not create confusion

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:294]: okay i changed it to a more incorrect option so it does not create confusion

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:298]: okay i changed it to a more incorrect option so it does not create confusion

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:240]: if the browser doesn't support variables you need a browser fallback, ie the same property above that doesn't use `var()`. See https://www.freecodecamp.org/learn/responsive-web-design/basic-css/improve-compatibility-with-browser-fallbacks for an example. Is this the type of fallback you are asking about?

Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:240]: Yes, that’s the type of fallback i mean. Providing a standard property value alongside the CSS variable ensures that the styles work in browsers that don't support variables.

Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:240]: it's a fallback also the second value in the `var()` function, maybe a different wording of the question would be more clear?

Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:240]: okay on it

Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:240]: What is the purpose of providing a fallback value in the var() function for CSS custom properties?

is this right?

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: Here’s a shorter, to-the-point version:

Hey! I couldn’t find any concrete sources specifically about issues with CSS variables and older layout models like float or table. CSS variables are generally well-supported, so browser compatibility is the main concern.

I’ll go ahead and update the quiz question to reflect that. Thanks for pointing it out!

Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: Hey! I couldn’t find any concrete sources specifically about issues with CSS variables and older layout models like float or table. CSS variables are generally well-supported, so browser compatibility is the main concern.

I’ll go ahead and update the quiz question to reflect that. Thanks for pointing it out!

File: curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md
Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md:240]: so you are changing the type of fallback you are asking about? if so make sure to also change the answer

Comment (Ayushjhawar8) [reply curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md]: done please check

**Comment by Ayushjhawar8** [comment]
@ilenia-magoni can you please tell why my last commit have that cross (X) mark on it?

**Comment by ilenia-magoni** [comment]
@Ayushjhawar8 the portuguese and italian builds had issues. Ignore that.

**Comment by Ayushjhawar8** [comment]
okay also please can you change the label i have done all the changes said till now

**Review by gikf — APPROVED** [review]

**Comment by ilenia-magoni** [comment]
Hi @Ayushjhawar8 . Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Comment by Ayushjhawar8** [comment]
Hello @naomi-lgbt . Please can you provide me hacktoberfest role in the discord server
my discord id:- aurorarift

Thanks you =)

**Pull request merged**
